### PR TITLE
ACI-CNI 5.2.3.2 upgrade

### PR DESCRIPTION
**Pull request by siva-muni**
- Added changes of ACI-CNI 5.2.3.3.

### CODE REVIEW

File: scripts/validate [hunk 1]
```diff
@@ -15,7 +15,7 @@ echo Verifying modules
 go mod verify
 
 echo Generating files
-go generate
+#go generate
```
Comment (rosskirkpat) [inline scripts/validate]: Why is the `go generate` commented out here?

File: data/data.json [hunk 2]
```diff
@@ -9146,19 +9146,19 @@
    "weaveNode": "weaveworks/weave-kube:2.8.1",
    "weaveCni": "weaveworks/weave-npc:2.8.1",
    "podInfraContainer": "rancher/mirrored-pause:3.6",
-   "ingress": "rancher/nginx-ingress-controller:nginx-1.2.1-rancher1",
+   "ingress": "rancher/nginx-ingress-controller:nginx-1.2.0-rancher1",
```
Comment (rosskirkpat) [inline data/data.json]: The data.json is pulled in during CI and should not be modified directly, instead the changes should be merged in KDM and then pulled through here.

File: cluster/defaults.go [hunk 3]
```diff
@@ -72,35 +72,52 @@ const (
...
+	DefaultEnableUpdates                        = "false"
+	DefaultMTUHeadroom                          = "None"
+	DefaultNodeSvcSubnet                        = "None"
+	DefaultDisableWaitForNetwork                = "false"
+	DefaultDurationWaitForNetwork               = "210"
+	DefaultPodSubnet                            = "None"
+	DefaultWatchNamespace                       = ""
+	DefaultUseAppsAPI                           = "apps/v1"
+	DefaultKubeConfigController                 = "1.1.1.1"
+	DefaultFlavor                               = "kubernetes-1.22"
```
Comment (rosskirkpat) [inline cluster/defaults.go]: Is there a reason you hard-coded to 1.22? Does ACI 5.2.3.2 support 1.23+ of kubernetes?

File: cluster/defaults.go [hunk 4]
```diff
@@ -72,35 +72,52 @@ const (
...
+	DefaultOpflexDeviceDeleteTimeout            = "None"
+	DefaultEnableUpdates                        = "false"
+	DefaultMTUHeadroom                          = "None"
+	DefaultNodeSvcSubnet                        = "None"
+	DefaultDisableWaitForNetwork                = "false"
+	DefaultDurationWaitForNetwork               = "210"
+	DefaultPodSubnet                            = "None"
+	DefaultWatchNamespace                       = ""
+	DefaultUseAppsAPI                           = "apps/v1"
+	DefaultKubeConfigController                 = "1.1.1.1"
```
Comment (rosskirkpat) [inline cluster/defaults.go]: What is the functionality of having this default to 1.1.1.1? I can see this causing issues with CNIs that are not ACI.

File: cluster/defaults.go [hunk 5]
```diff
@@ -72,35 +72,52 @@ const (
...
+	DefaultAciRunGbpContainer                   = "false"
+	DefaultAciRunOpflexServerContainer          = "false"
+	DefaultAciUseAciAnywhereCRD                 = "false"
+	DefaultAciEnableEndpointSlice               = "false"
+	DefaultAciOpflexClientSSL                   = "true"
+	DefaultAciUsePrivilegedContainer            = "false"
+	DefaultAciUseOpflexServerVolume             = "false"
+	DefaultApicSubscriptionDelay                = "None"
+	DefaultApicRefreshtickerAdjust              = "None"
+	DefaultOpflexDeviceDeleteTimeout            = "None"
```
Comment (rosskirkpat) [inline cluster/defaults.go]: At a high level - all of these new defaults should be tucked under `DefaultAci`* if they are only being added for the ACI-CNI

**Comment by siva-muni** [comment]
[rosskirkpat](https://github.com/rosskirkpat), Thanks for the review. we will address your comments.

**Comment by siva-muni** [comment]
We are refactoring the code, removed the knobs and variables as not needed closing the PR for now. We will re open if there is requirement for this knobs.

**Comment by snasovich** [comment]
@siva-muni , thank you for updating this PR. Please reply to comments left by @rosskirkpat so we have more context into how they were addressed (or if they are not addressed yet).

File: scripts/validate (hunk 1 above)
Comment (siva-muni) [reply scripts/validate]: commented for internal testing, revert back the changes in [4997744](https://github.com/rancher/rke/pull/2963/commits/49977444ef6dbca80e05bb1282e7330ae7dc9b90)

File: data/data.json (hunk 2 above)
Comment (siva-muni) [reply data/data.json]: Revert back the changes [4997744](https://github.com/rancher/rke/pull/2963/commits/49977444ef6dbca80e05bb1282e7330ae7dc9b90)

File: cluster/defaults.go (hunk 3 above)
Comment (siva-muni) [reply cluster/defaults.go]: while refactoring the code, removed this variable in[4997744](https://github.com/rancher/rke/pull/2963/commits/49977444ef6dbca80e05bb1282e7330ae7dc9b90) No it doesn't support 1.23+

File: cluster/defaults.go (hunk 4 above)
Comment (siva-muni) [reply cluster/defaults.go]: while refactoring the code, removed this variable in [4997744](https://github.com/rancher/rke/pull/2963/commits/49977444ef6dbca80e05bb1282e7330ae7dc9b90)

File: cluster/defaults.go (hunk 5 above)
Comment (siva-muni) [reply cluster/defaults.go]: updated as DefaultAci* in [4997744](https://github.com/rancher/rke/pull/2963/commits/49977444ef6dbca80e05bb1282e7330ae7dc9b90)

**Comment by siva-muni** [comment]
[snasovich](https://github.com/snasovich), Thank you, I replied to the comments.

**Comment by snasovich** [comment]
@siva-muni , thank you. I'll let @rosskirkpat and @kinarashah lead the review on this one, but to me the changes look sane.
Please note that @rosskirkpat is out today so it's unlikely we'll hear back from him on this today.

**Comment by siva-muni** [comment]
We are finished our testing, it was successful. We are not seeing any issues.

**Review by rosskirkpat — APPROVED** [review]
LGTM, thank you for addressing the feedback.

**Comment by rosskirkpat** [comment]
@siva-muni On the note of ACI not supporting >1.22 clusters, we can potentially block the ability to choose ACI as the CNI if the kubernetes version is >1.22. Is that something you would be interested in?

**Comment by snaiksat** [comment]
@rosskirkpat thanks for bringing this up. We will be supporting k8s 1.23 shortly, its just that the support has not been released (we are going through some testing/validation at our end). I anticipate the support for k8s 1.23 to be made available in an upcoming ACI-CNI patch release available by the end of this month. So I would ask you a slightly different question, would we have an opportunity to post a follow-up PR targeted for v1.3 to reflect our support for k8s 1.23? Or should we just proactively include support for the version right away (I would still prefer that we merge the two PRs which are in review and we can quickly post a follow up PR to include k8s 1.23 too).

**Comment by LavanyaVaddavalli** [comment]
@rosskirkpat 

We have added 5.2.3.2 images to 1.23.8-rancher1-1 and 1.24.2-rancher1-1 versions as well and updated the PR in kontainer-driver-metadata repo. Currently 5.2.3.2 cannot support kubernetes 1.23+ versions. We will raise a new PR to support 1.23+ k8s versions with 5.2.3.3 aci-cni release.

**Comment by kinarashah** [comment]
@siva-muni Now that changes in KDM are merged, could you run `go generate` and push updated `data/` as a separate commit?

**Review by kinarashah — APPROVED** [review]

**Pull request merged**